include/alpaka/core/CallbackThread.hpp

```diff
@@ -16,10 +19,40 @@ namespace alpaka::core
 {
     class CallbackThread
     {
-        // std::packaged_task is used because std::function requires that the wrapped callable is copyable.
+#if BOOST_COMP_CLANG
+#    pragma clang diagnostic push
+#    pragma clang diagnostic ignored "-Wweak-vtables"
+#endif
+        // A custom class is used because std::function<F> requires F to be copyable, and std::packed_task provides a
+        // std::future which will keep the task alive and we can not control the moment the future is set.
```

Comment:
```suggestion
        // A custom class is used because std::function<F> requires F to be copyable, and std::packaged_task provides a
        // std::future which will keep the task alive and we cannot control the moment the future is set.
```